### PR TITLE
More stable computation of KL between two Bernoulli distributions

### PR DESCRIPTION
**Pull request by nicola-decao**
I propose to use `Bernoulli.logits` instead of `Bernoulli.probs` when computing the KL between two Bernoulli distributions. The two formulations are equivalent but since usually neural networks output logits, the second is more stable.

### CODE REVIEW

File: torch/distributions/kl.py
Comment (vishwakftw) [inline torch/distributions/kl.py]: This is slightly inefficient I think. For numerical stability, you can try rewriting the entire expression as:
`p.probs * p.logits - p.probs * q.logits + (-p.probs).log1p() - (-q.probs).log1p()`.

However, if you are specifying `logits` alone, then the `logits_to_probs` will be called when the `probs` attribute is used.

Comment (nicola-decao) [reply torch/distributions/kl.py:179]: `p.probs * (p.logits - q.logits) + (-p.probs).log1p() - (-q.probs).log1p()` is even better. Besides we call `.probs` many times anyway. But in this way should be more stable.

**Comment by nicola-decao** [comment]
@vishwakftw are you fine with my comment? I can update the PR if you think we should go through. I never use the current implementation in my code as it goes to NaN for many cases where it doesn't if you use logits.

**Comment by fritzo** [comment]
LGTM, thanks for stabilizing this @nicola-decao !

**Review by fritzo — APPROVED** [review]

**Comment by nicola-decao** [comment]
@fritzo what is blocking this PR?

**Comment by fritzo** [comment]
> @fritzo what is blocking this PR?

I no longer have merge rights, so we need to ask a Meta employee.  Actually all my recent PR approvals seem to have gone stale.

@ezyang @neerajprad would either of you be able to merge this?

**Review by neerajprad — APPROVED** [review]

**Comment by neerajprad** [comment]
> @nicola-decao I no longer have merge rights, so we need to ask a Meta employee. Actually all my recent PR approvals seem to have gone stale.

@ezyang: Not sure if there was some policy change, but is it possible to restore merge rights to OS contributors who have created/are maintaining the module?

**Comment by neerajprad** [comment]
@nicola-decao: The base revision is stale, do you mind rebasing your diff onto master and pushing again?

**Comment by ezyang** [comment]
@neerajprad yeah, let me figure out what went wrong

**Pull request closed without merging**
